### PR TITLE
feat: [code-first] building the backend api

### PR DESCRIPTION
**Pull request by MateuszMyrcik**
### **Grading Checklist**
- [x]  I have built an MVC RESTful API using Express.js and TypeScript.
- [x] I have included the following API endpoints:
POST to /users/new: Creates a new user.
POST to /users/edit/:userId: Edits a user.
GET to /users?email=someuser@gmail.com: Fetches a user by their email
- [x] I should be able to call each endpoint passing in data which triggers each of the non-500 status code success and error edge cases.
- [x] I have chosen either a relational database (preferably Postgres) or a NoSQL database (such as MongoDB) to store user data.
- [x] (Optionally), I used an ORM (e.g., TypeORM or Prisma) for managing data in the chosen database.
- [x]  I can manually test and verify that the API endpoints work using Postman or CURL.

### CODE REVIEW

File: backend/src/user/Controller.ts [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+
+export class UserController {
+  private userModel: UserModel;
+
+  constructor(userModel: UserModel) {
+    this.userModel = userModel;
+  }
+
+  async createUser(req: express.Request, res: express.Response) {
+    const { username, email, firstName, lastName } = req.body;
```
Comment (acsauk) [inline backend/src/user/Controller.ts:23]: Nice and clear here for expectations of the body contents 👍

File: backend/src/user/Model.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    });
+  }
+
+  async editUser(user: Omit<User, "password" | "id"> & { id: number }) {
+    const { email, firstName, lastName, username } = user;
+    return this.db.user.update({
+      where: { id: user.id },
+      data: { email, firstName, lastName, username },
+    });
+  }
```
Comment (acsauk) [inline backend/src/user/Model.ts:36]: Good encapsulation of model functionality - its clear what each query is for.

**Review by acsauk — APPROVED** [review]
Not really much to add - this approach hits all of the grading checkpoint items and some initial stages of separation of concerns which for this level is some bonus points I guess 😅

**Comment by MateuszMyrcik** [comment]
> Not really much to add - this approach hits all of the grading checkpoint items and some initial stages of separation of concerns which for this level is some bonus points I guess 😅

Needed to block myself a bit in some places,  wanted to go further😅

**Comment by acsauk** [comment]
Same!

**Pull request merged**